pylabrobot/resources/carrier.py

```diff
@@ -27,14 +29,22 @@ def assign_child_resource(
     reassign: bool = True
   ):
     self.resource = resource
-    return super().assign_child_resource(resource, location=location)
+    return super().assign_child_resource(resource, location, reassign)
 
   def unassign_child_resource(self, resource):
     self.resource = None
     return super().unassign_child_resource(resource)
+  
+  def serialize(self) -> dict:
+      return { **super().serialize(), "name": self.name}
```

Comment:
"name" shoudl already be serialized in super, so this function does not need overriding.